vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts

```diff
@@ -77,26 +78,33 @@ export class InstallTrackerSingleton
 
     protected executeWithLock = async <A extends any[], R>(alreadyHoldingLock : boolean, f: (...args: A) => R, ...args: A): Promise<R> =>
     {
-
         const trackingLock = 'tracking.lock';
         const lockPath = path.join(__dirname, trackingLock);
         fs.writeFileSync(lockPath, '', 'utf-8');
 
+        let returnResult : any;
+
         this.eventStream?.post(new DotnetLockAttemptingAcquireEvent(`Lock Acquisition request to begin.`, new Date().toISOString(), lockPath, lockPath));
         try
         {
             if(alreadyHoldingLock)
             {
                 return await f(...(args));
             }
-            const release = await lockfile.lock(lockPath, { retries: { retries: 10, minTimeout: 5, maxTimeout: 2000 } });
-            try
-            {
-                return await f(...(args));
-            }
-            finally
+            else
             {
-                await release();
+                await lockfile.lock(lockPath, { retries: { retries: 10, minTimeout: 5, maxTimeout: 10000 } })
```

Comment:
What units are the min and max timeout in?

Reply:
they are in MS. The library will do an exponential equation where it starts trying again after 5 ms and then takes longer each retry until its at 10000 ms.